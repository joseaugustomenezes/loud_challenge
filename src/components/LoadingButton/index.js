import React from "react";
import Button from "react-bootstrap/Button";
import Spinner from "react-bootstrap/Spinner";

const LoadingButton = ({ loading, children, ...rest }) => (
  <Button {...rest} disabled={loading}>
    {loading && (
      <Spinner style={{ marginRight: '5px' }} animation="border" size="sm" role="status">
        <span className="sr-only">Loading...</span>
      </Spinner>
    )}
    {children}
  </Button>
);

export default LoadingButton;
