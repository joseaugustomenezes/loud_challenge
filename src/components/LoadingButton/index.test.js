import React from 'react';
import { shallow, mount } from 'enzyme';

import LoadingButton from './index';

describe('<LoadingButton />', () => {
  it('should render correctly', () => {
    const component = shallow(<LoadingButton />);
    expect(component).toMatchSnapshot();
  });

  it('should not call onclick function ', () => {
    const clickFn = jest.fn();

    const component = mount(<LoadingButton loading={true} onClick={() => clickFn()} />);
    component.find('button').simulate('click');
    expect(clickFn).not.toHaveBeenCalled();
    component.unmount();
  });

  it('should call onclick function', () => {
    const clickFn = jest.fn();

    const component = mount(<LoadingButton loading={false} onClick={() => clickFn()} />);
    component.find('button').simulate('click');
    expect(clickFn).toHaveBeenCalled();
    component.unmount();
  });
});