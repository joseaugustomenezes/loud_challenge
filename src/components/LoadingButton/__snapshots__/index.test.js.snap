// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingButton /> should render correctly 1`] = `
<Button
  active={false}
  disabled={false}
  variant="primary"
/>
`;
